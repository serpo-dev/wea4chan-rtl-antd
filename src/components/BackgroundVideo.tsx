import { FC } from "react";
import bg720p from "../assets/720p.mp4";

export const BackgroundVideo: FC = () => {
    return (
        <video
            autoPlay
            muted
            loop
            style={{
                position: "fixed",
                objectFit: "cover",
                width: "100vw",
                height: `calc(100vh - 160px)`,
                pointerEvents: "none",
                filter: "brightness(49%)",
            }}
            className="undraggable unselectable"
        >
            <source
                src={bg720p}
                type="video/webm"
            />
            Your browser does not dupport HTML5 video
        </video>
    );
};
