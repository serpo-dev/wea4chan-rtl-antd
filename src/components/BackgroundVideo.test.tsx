import { render } from "@testing-library/react";
import { BackgroundVideo } from "./BackgroundVideo";
import "@testing-library/jest-dom";

describe("should render a valid background video", () => {
    it("snapshot should match to the component render view", () => {
        const { asFragment } = render(<BackgroundVideo />);
        expect(asFragment()).toMatchSnapshot();
    });
});
