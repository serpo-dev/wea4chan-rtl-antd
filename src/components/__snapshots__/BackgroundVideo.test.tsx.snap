// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a valid background video snapshot should match to the component render view 1`] = `
<DocumentFragment>
  <video
    autoplay=""
    class="undraggable unselectable"
    loop=""
    style="position: fixed; object-fit: cover; width: 100vw; height: calc(100vh - 160px); pointer-events: none; filter: brightness(49%);"
  >
    <source
      src="720p.mp4"
      type="video/webm"
    />
    Your browser does not dupport HTML5 video
  </video>
</DocumentFragment>
`;
